src/HttpHeader.cc

```diff
@@ -522,18 +524,40 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
                Raw("header", header_start, hdrLen));
     }
 
+    String rawTe;
     if (clen.prohibitedAndIgnored()) {
+        // NOTE: prohibitedAndIgnored() includes trailer header blocks
+        //       being parsed as a case to forbid/ignore these headers.
+
```

Comment:
Nit: This code is not the place where we should document what prohibitedAndIgnored() means or what cases it covers. I suggest just dropping this comment (and the following new line -- GitHub does not support that via suggestions). If you prefer, let's improve ContentLengthInterpret API documentation to detail this.

Reply:
This is here to mark specifically prohibitedAndIgnored() implications on the following delById() calls extending beyond the RFC requirements also commented here. It is unrelated to any other uses of prohibitedAndIgnored().

Reply:
> This is here to mark specifically prohibitedAndIgnored() implications on the following delById() calls extending beyond the RFC requirements

The source code comment about the RFC requirements suffers from exactly the same problem -- it is misplaced. This PR makes things arguably worse by adding another misplaced comment to an existing misplaced comment (using that existing misplaced comment as the hidden/undisclosed reason/excuse). See my intial [suggestion](https://github.com/squid-cache/squid/pull/702#discussion_r465177177) for two ways to avoid aggravating this problem.

Please note that I am just explaining my (optional) change suggestion. I am not blocking this PR because it adds a misplaced comment.

Reply:
Also, if you insist on adding this misplaced comment, please at least remove its `NOTE: ` prefix. It is just noise  -- virtually all comments are "please note: ..." code annotations.

Reply:
Removed the "NOTE:" prefix in 50643ac0d.